amundsen_application/static/js/components/TableDetail/ColumnStats/index.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import * as moment from 'moment-timezone';
+
+import { TableColumnStats } from 'interfaces/index';
+
+import './styles.scss';
+
+export interface ColumnStatsProps {
+  stats: TableColumnStats[];
+}
+
+export class ColumnStats extends React.Component<ColumnStatsProps> {
+  constructor(props) {
+    super(props);
+  }
+
+  formatDate = (unixEpochSeconds) => {
+    return moment(unixEpochSeconds * 1000).format("MMM DD, YYYY");
+  };
+
+  getStatsInfoText = (startEpoch: number, endEpoch: number) => {
+    const startDate = startEpoch ? this.formatDate(startEpoch) : null;
+    const endDate = endEpoch ? this.formatDate(endEpoch) : null;
+
+    let infoText = 'Stats reflect data collected';
+    if (startDate && endDate) {
+      if (startDate === endDate) {
+        infoText = `${infoText} on ${startDate} only. (daily partition)`;
+      } else {
+        infoText = `${infoText} between ${startDate} and ${endDate}.`;
+      }
+    } else {
+      infoText = `${infoText} over a recent period of time.`;
+    }
+    return infoText;
+  };
+
+  renderColumnStat = (entry: TableColumnStats) => {
+    return (
+      <div className="column-stat-row" key={entry.stat_type}>
+        <div className="stat-name body-3">
+          { entry.stat_type.toUpperCase() }
+        </div>
+        <div className="stat-value">
+          { entry.stat_val }
+        </div>
+      </div>
+    )
+  };
+
+  render = () => {
+    const { stats } = this.props;
+    if (stats.length === 0) {
+      return null;
+    }
+    const startEpoch = Math.min(...stats.map(s => s.start_epoch));
+    const endEpoch = Math.max(...stats.map(s => s.end_epoch));
+
+    return (
```

Comment:
nit: some decoupling will help to write better tests here. We can also write tests that better verify what goes on in those `map` statements. Exact syntax might need tweaking, essentially raising for the future that we can assert that a thing happened for each thing, in exactly the way that it was supposed to.  
```
props.stats.forEach((state, index ) => {
    expect(renderColumnStatSpy).toHaveBeenCalledWith(stat);
});
```
If the logic we are testing has conditional logic, then that logic can also be separated out  in the code and mocked in the test.

Reply:
Added a TODO to move map functions, also added test case for handling each stat.